### PR TITLE
Gitlab support

### PR DESCRIPTION
**Pull request by cotocisternas**
## Description

- Add gitlab support for the create_gitops_prs command
- Document git_server options and params

## Related Issue

#68

## Motivation and Context

Extend the gitops integration support

## How Has This Been Tested?

Clean run of bazel test

New test has been added in `gitops/git/gitlab/gitlab_test.go`

```go
package gitlab

import "testing"

func TestCreatePRRemote(t *testing.T) {
	t.Skip("Manual")
	var (
		testGitlabToken = "********"
	)
	accessToken = &testGitlabToken
	type args struct {
		from  string
		to    string
		title string
	}
	tests := []struct {
		repo    string
		args    args
		wantErr bool
	}{
		{
			repo: "cotocisternas/rules_gitops_gitlab_test",
			args: args{
				from:  "feature/gitlab-test",
				to:    "master",
				title: "test_gitlab",
			},
			wantErr: false,
		},
		{
			repo: "petabytecl/subgroup_rules_gitops_gitlab_test/rules_gitops_gitlab_test",
			args: args{
				from:  "feature/gitlab-test",
				to:    "master",
				title: "test_gitlab",
			},
			wantErr: false,
		},
	}
	for _, tt := range tests {
		t.Run(tt.repo, func(t *testing.T) {
			repo = &tt.repo
			if err := CreatePR(tt.args.from, tt.args.to, tt.args.title); (err != nil) != tt.wantErr {
				t.Errorf("CreatePR() error = %v, wantErr %v", err, tt.wantErr)
			}
		})
	}
}
```

[User test case MRs](https://gitlab.com/cotocisternas/rules_gitops_gitlab_test/-/merge_requests?scope=all&utf8=%E2%9C%93&state=all)
[Group test case MRs](https://gitlab.com/petabytecl/subgroup_rules_gitops_gitlab_test/rules_gitops_gitlab_test/-/merge_requests?scope=all&utf8=%E2%9C%93&state=all)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: gitops/git/gitlab/gitlab.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+var (
+	gitlabHost  = flag.String("gitlab_host", "https://gitlab.com", "The host name of the gitlab instance")
+	repo        = flag.String("gitlab_repo", "", "the repo to use for gitlab api requests")
+	accessToken = flag.String("gitlab_access_token", os.Getenv("GITLAB_TOKEN"), "the access token to authenticate requests")
+)
+
+func CreatePR(from, to, title string) error {
+	if *accessToken == "" {
+		return errors.New("github_access_token must be set")
```
Comment (kzadorozhny) [inline gitops/git/gitlab/gitlab.go:22]: The message should be `"gitlab_access_token ... "`

File: gitops/git/gitlab/gitlab.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	}
+
+	createdPr, resp, err := gl.MergeRequests.CreateMergeRequest(*repo, &opts)
+	if err == nil {
+		log.Println("Created MR: ", createdPr.WebURL)
+		return nil
+	}
+
+	if resp.StatusCode == http.StatusConflict {
+		// Handle the case: "Create PR" request fails because it already exists for this source branch
```
Comment (kzadorozhny) [inline gitops/git/gitlab/gitlab.go:53]: To be consistent, we should probably stick to the Gitlab terminology here: "MR" not "PR".

**Review by kzadorozhny — CHANGES_REQUESTED** [review]
Overall LGTM, but the are minor tweaks.

**Review by kzadorozhny — DISMISSED** [review]
Thank you for the contribution!

File: README.md [hunk 3]
```diff
@@ -58,10 +58,10 @@ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 rules_gitops_version = "ef7a846a58e3baeaceb7067c05f17fc3e8a821fd"
 
 http_archive(
-    name = "com_adobe_rules_gitops",
-    sha256 = "19a394fe36cd0ac21d0df1357a5e9affbd45c61c66f76fca366d3899b2f1f2d6",
-    strip_prefix = "rules_gitops-%s" % rules_gitops_version,
-    urls = ["https://github.com/adobe/rules_gitops/archive/%s.zip" % rules_gitops_version],
+  name = "com_adobe_rules_gitops",
```
[the remark this answers is not in the document]
Comment (cotocisternas) [reply README.md]: unintended markdown reformat, commit [f219869](https://github.com/adobe/rules_gitops/pull/71/commits/f219869d33822012395670327cd599361c511173) fixes

File: gitops/git/gitlab/gitlab.go (hunk 1 above)
Comment (cotocisternas) [reply gitops/git/gitlab/gitlab.go]: done

File: gitops/git/gitlab/gitlab.go (hunk 2 above)
Comment (cotocisternas) [reply gitops/git/gitlab/gitlab.go]: done

**Review by kzadorozhny — APPROVED** [review]

**Pull request merged**